src/model/immutable/EditorBidiService.js

```diff
@@ -34,16 +34,30 @@ const EditorBidiService = {
     }
 
     const blockMap = content.getBlockMap();
-    const nextBidi = blockMap
-      .valueSeq()
-      .map(block => nullthrows(bidiService).getDirection(block.getText()));
-    const bidiMap = OrderedMap(blockMap.keySeq().zip(nextBidi));
 
-    if (prevBidiMap != null && Immutable.is(prevBidiMap, bidiMap)) {
+    let needsNewBidiMap = !prevBidiMap;
+    if (!needsNewBidiMap) {
+      needsNewBidiMap = blockMap.count() !== prevBidiMap.count();
+    }
+    if (!needsNewBidiMap) {
+      needsNewBidiMap = Boolean(
+        blockMap.find((block: ContentBlock, key: string) => {
+          return (
+            !prevBidiMap.has(key) ||
+            prevBidiMap.get(key) !==
```

Comment:
does prevBidiMap.get() return undefined? do you need the has?

Reply:
It would return undefined if it didn't have the key, but that's not sufficient. I didn't want to make assumptions about the types of the values dtored in `prevBidiMap`.